docs/src/content/docs/book/functions.mdx

```diff
@@ -193,10 +472,544 @@ contract ManualMethodId {
+
+<Badge text="Any scope" title="This attribute can be applied to functions within contracts and functions at the top level" variant="note" size="medium"/><p/>
+
+The `inline{:tact}` attribute embeds the code of the function to its call sites, which eliminates the overhead of the calls but potentially increases the code size if the function is called from multiple places.
+
+This attribute can improve performance of large infrequently called functions or small, alias-like functions. It can be combined with any other attribute except for [`get{:tact}`](#attributes-get).
+
+```tact
+inline fun add(a: Int, b: Int) {
+    // The following addition will be performed in-place,
+    // at all the `add()` call sites.
+    return a + b;
+}
+```
+
+#### `get` {#attributes-get}
+
+<Badge text="Contract scope" variant="note" size="medium"/><p/>
+
+The special `get{:tact}` attribute cannot be combined with any other attribute. It is applied to [internal functions](#fun-internal), transforming them into [getter functions](#get).
+
+## Commonalities
+
+Functions commonly share the same aspects, parts, or behavior. For example, due to the nature of TVM, Tact uses the [call by value](https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_value) parameter-passing and binding strategy, which applies to all function kinds. That is, the evaluated value of any variable passed in a function call or assigned in the [`let{:tact}`](/book/statements#let) or [assignment](/book/statements#assignment) statement is copied.
+
+Therefore, there are no modifications of the original values in different scopes, except for the [extension _mutation_ functions](#attributes-mutates). And even so, those functions do not mutate the original value denoted by the `self{:tact}` keyword — instead, they discard the old value and replace it with the new one obtained at the end of the function body.
+
+The above applies to all functions except for [receivers](#receivers), because they cannot be called directly and are only executed upon receiving specific messages.
+
+### Scope {#commonalities-scope}
+
+All kinds of functions vary in scope: some can only be defined at the top level, such as [global functions](#fun-global), while others can only be defined within contracts and traits, such as [internal functions](#fun-internal) or [receivers](#receivers).
+
+Functions are hoisted upon declaration, such that the ones declared earlier might call those declared later and vice versa.
+
+```tact
+// Global function first() is defined prior to the second() function,
+// but we can call the second() one just fine.
+fun first(): Bool {
+    return second(true); // always true
+}
+
+// Global function second() is defined after the first() function,
+// and call to the first() function works as expected.
+fun second(flag: Bool): Bool {
+    if (flag) {
+        return true;
+    }
+    return first(); // always true, because of the prior condition
+}
+```
+
+### Return values {#commonalities-return}
```

Comment:
Link to the explanation that it should be obvious to the compiler that returns are always reachable